docs/networks/mainnet/lukso-mainnet.md

```diff
@@ -0,0 +1,228 @@
+
+[https://style-landing-page-cu-33a6bb.network-deposit-launchpad-private.pages.dev/en/](https://style-landing-page-cu-33a6bb.network-deposit-launchpad-private.pages.dev/en/)
+
+<--! TODO: This link above is broken-->
+
+### Key Generation via CLI
+
+We also provide a Command Line tool for you and strongly suggest that you do it in an air-gapped computer and store your keys safely.
+
+The program generates a set of validator keys as .json files which you will then need to deposit to run your validator.
+
+- Step 1: In this link: [](https://github.com/lukso-network/tools-key-gen-cli/releases) please download the latest version of the Linux / MacOs version of the LUKSO CLI to your computer
+
+#### Linux-amd64
+
+```console
+curl -O https://github.com/lukso-network/tools-key-gen-cli/releases/download/v2.4.7/lukso-key-gen-cli-v2.4.7-linux-amd64.tar.gz
+```
+
+#### MacOs-amd64
+
+```console
+curl -O https://github.com/lukso-network/tools-key-gen-cli/releases/download/v2.4.7/lukso-key-gen-cli-v2.4.7-macos-amd64.tar.gz
+```
+
+#### Windows-amd64
+
+```console
+curl -O https://github.com/lukso-network/tools-key-gen-cli/releases/download/v2.4.7/lukso-key-gen-cli-v2.4.7-windows-amd64.tar.gz
+```
+
+<--! TODO: Get the proper link to the KeyGen deposit-->
+
+This CLI is pretty straightforward and allows only 2 types of commands: an existing-mnemonic (in case you already have one and would only need to regenerate the .json file with your keys) or a new-mnemonic, which generates a brand new set of keys to your validator.
+
+After downloading and installing the tool, you should be able to find it at:
+
+```
+/usr/local/bin/lukso-keygen
+```
+
+#### Generating a NEW mnemonic:
+
+Simply type:
+
+```console
+lukso-keygen new-mnemonic
+```
+
+and follow the instructions (for English, press the "ENTER" key twice):
+chose the number of validators you want to run:
+chose the network/chain name: 'lukso', 'l16', 'l2022'
+[lukso]: l2022
+create & retype a strong password
+
+And finally you will have your mnemonic (12 keywords written in English, or any other language you selected before).
+
+Please copy/paste them; write them down, store them on a safe place.
+
+You will be asked to re-type each of those 12 words in order; until you see a Rhino ASCII art drawing.
+
+#### Generating a key from an EXISTING mnemonic:
+
+It might be that you already have a set of 12 words that you want to use to generate the Keys again.
+
+In this case:
+
+```console
+lukso-keygen existing-mnemonic --eth1_withdrawal_address <0xADDRESS_YOU_HAVE>
+```
+
+and follow the instructions.
+
+Your keys will be stored in the directory:
+
+```
+/Users/username/validator_keys
+```
+
+### Key Generation CLI with Docker
+
+Running the following command will allow you to install and generate a new set of keys using docker
+
+```console
+ docker run -it -v $(pwd)/validator_keys:/app/validator_keys --rm europe-docker.pkg.dev/lks-lz-artifacts/docker-key-gen-cli/lukso-key-gen-cli new-mnemonic --help
```

Comment:
The docker commando line also allows a :v2.4.7 docker tag. New releases will automatically update the :latest tag which is used by default if no tag is specified